### PR TITLE
Fixing ACR credential issues when up command was used

### PR DESCRIPTION
**Pull request by harryli0108**
Fixing the issue where ACR authentication fails when using up command by swapping `up` with `containerapp update` and added `ingress update`

### CODE REVIEW

**Comment by harryli0108** [comment]
Please don't merge before team member approved this change.

File: azurecontainerapps.ts
Comment (cormacpayne) [inline azurecontainerapps.ts:612]: I would be in favor of us having two separate calls here: one that updates the Container App and another that updates the ingress for the Container App

```typescript
// Update the Container App using the 'containerapp update' command
await this.appHelper.updateContainerAppWithUpdate(this.containerAppName, this.resourceGroup, this.commandLineArgs);

// Update the ingress and target port on the Container App using the 'ingress update' command
await this.appHelper.updateContainerAppIngress(this.containerAppName, this.resourceGroup, this.ingress, this.targetPort);
```

This will allow the functions to not be overloaded with functionality and be reusable in other scenarios we may want to use them in.

**Review by cormacpayne — CHANGES_REQUESTED** [review]

File: azurecontainerapps.ts
Comment (harryli0108) [reply azurecontainerapps.ts:612]: makes sense! changing this in the next commit

File: azurecontainerapps.ts
Comment (snehapar9) [inline azurecontainerapps.ts:612]: Synced offline, since we don't use the `up` command to update the container app anymore, we should change the name of the global variable `this.shouldUseUpdateCommand` to clearly indicate when the `update ingress` command needs to be executed.

File: src/ContainerAppHelper.ts
Comment (snehapar9) [inline src/ContainerAppHelper.ts:154]: Can we remove `optionalCmdArgs` on line 154? I'm not seeing it in the method `updateContainerAppIngress` below?

File: azurecontainerapps.ts
Comment (harryli0108) [reply azurecontainerapps.ts:617]: sounds good! updated in the latest commit

File: src/ContainerAppHelper.ts
Comment (harryli0108) [reply src/ContainerAppHelper.ts]: makes sense. fixed in the latest commit.

File: .github/workflows/run-validation-oidc.yml
Comment (cormacpayne) [inline .github/workflows/run-validation-oidc.yml:603]: What's the reason for this change? It looks like other tests are still referencing `env` and not `var`

File: src/ContainerAppHelper.ts
Comment (cormacpayne) [inline src/ContainerAppHelper.ts:151]: nit: it would be great to re-word this comment:

_Updates the ingress on an existing Container App. Note: this helps to avoid failure during ACR authentication._

File: azurecontainerapps.ts
Comment (cormacpayne) [inline azurecontainerapps.ts:513]: nit: let's leave a comment above this variable specifying what it represents and why we're setting it to `true` in this case

File: azurecontainerapps.ts
Comment (cormacpayne) [reply azurecontainerapps.ts:617]: @snehapar9 -- I'm not sure I understand your comment _since we don't use the `up` command to update the container app anymore_ because I do see a function call a few lines up `updateContainerAppWithUp` that still uses `up`. Would you mind elaborating a bit more?

**Review by cormacpayne — APPROVED** [review]

File: .github/workflows/run-validation-oidc.yml
Comment (harryli0108) [reply .github/workflows/run-validation-oidc.yml:603]: This one was added because for some reason the action couldn't read the container app name (shown as empty in debug message). It then got fixed after I changed it from env to vars.

File: src/ContainerAppHelper.ts
Comment (harryli0108) [reply src/ContainerAppHelper.ts:151]: Sounds good! I'll have a separate pr to update the comments

File: azurecontainerapps.ts
Comment (snehapar9) [reply azurecontainerapps.ts:617]: @cormacpayne This comment does not need to be addressed anymore.
@harryli0108's initial changes had `up` removed and made calls to only `update`. But since `update` does not support registry credentials, `up` was added back.

**Pull request merged**
